compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1236,6 +1236,37 @@ impl<'a> Parser<'a> {
 
     /// Parse an indexing expression `expr[...]`.
     fn parse_index_expr(&mut self, lo: Span, base: P<Expr>) -> PResult<'a, P<Expr>> {
+        if self.prev_token.is_ident()
+            && matches!(self.token.kind, token::OpenDelim(Delimiter::Bracket))
+            && self.look_ahead(1, |t| t.is_lit())
+            && self.look_ahead(2, |t| matches!(t.kind, token::Semi))
+            && self.look_ahead(3, |t| t.is_lit())
+            && self.look_ahead(4, |t| matches!(t.kind, token::CloseDelim(Delimiter::Bracket)))
+        {
+            if let token::Ident(symbol, _) = self.prev_token.kind {
```

Comment:
You can add this to the `&&` chain above. If this is not allowed, then enable the `let_chains` feature in this crate.